### ISSUE TITLE
Incorrect Definition Name in secondary classes

### ISSUE THREAD

**Issue by luisjones** [json-everything/json-everything#769]
### Nuget Package

JsonSchema.Net.Generation

### Package Version

4.3.0.2

### Operating System

Mac

### .Net Target (if relevant, please specify the version in the description)

None

### A clear and concise description of the bug.

When having a [Description] attribute above a nested class that is not at the top level, Schema generation is creating a reference and definition called 'string' (and 'string1' for additional nested classes with descriptions) instead of nesting the description.

### What did you expect?

Actual:
```json
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "TestClass",
  "$id": "TestClass",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Test Name"
    },
    "nestedClass": {
      "type": "object",
      "properties": {
        "additionalProperty": {
          "$ref": "#/$defs/string"
        }
      },
      "description": "A Nested Class"
    }
  },
  "$defs": {
    "string": {
      "type": "string",
      "description": "An additional property"
    }
  }
}
```

Expected:

```json
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "TestClass",
  "$id": "TestClass",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Test Name"
    },
    "nestedClass": {
      "type": "object",
      "properties": {
        "additionalProperty": {
          "type": "string",
          "description": "An additional property"
        }
      },
      "description": "A Nested Class"
    }
  },
  "description": "A Nested Class"
}
```
I'm not concerned about there creating a def, but I would prefer if it would be named 'additionalProperty' instead of string.

### Please add test code or steps to reproduce the behavior.

The following does not work as expected:
```csharp
public class TestClass
{
    [Description("Test Name")]
    public string Name { get; set; }

    [Description("A Nested Class")]
    public TestNestedClass NestedClass { get; set; }
}

public class TestNestedClass
{
    [Description("An additional property")]
    public string AdditionalProperty { get; set; }
}
```
But this does:

```csharp
public class TestClass
{
    [Description("Test Name")]
    public string Name { get; set; }

    public TestNestedClass NestedClass { get; set; }
}

[Description("A Nested Class")]
public class TestNestedClass
{
    [Description("An additional property")]
    public string AdditionalProperty { get; set; }
}
```

### Code of Conduct & Contributor Guidelines

- [X] I agree to follow this project's Code of Conduct and Contribution Guidelines.

**Comment by gregsdennis** [comment]
Hey there.  Thanks for the report.  I don't think that I'd use "nested" class here.  That to me describes the scenario expressed in #767.  I'd call this a secondary class: it's just another class that's being used as a property.  In JSON Schema terms, it creates a "subschema", but I think this representation is where you've come up with the "nested" term.

`$ref`s are created as part of the optimization step, and (importantly) it has little to do with the classes at this point.  Instead, the optimization logic is just looking for commonly used subschemas and refactoring them into a definition.  Since schemas are generated from types, the type name is used as the definition name.  It has no reference to the property that used that type, so I can't use the property name as the definition name.  Moreover, I'm not sure it makes sense to since the extraction should only occur if the subschema is used in multiple places, meaning it could be part of multiple, differently-named properties.

In short, the type name is more appropriate for the definition name.

All of that said, I'm apparently having some problems with how `description` keywords are being applied, and I'm tracking that in #767.

**Comment by luisjones** [comment]
Thanks, I'm going to close this in favour of tracking in #767
